chainerx_cc/chainerx/array_repr.cc

```diff
@@ -238,16 +200,30 @@ struct ArrayReprImpl {
     }
 
 private:
-    template <typename T, typename Visitor>
-    void VisitElements(const Array& array, Visitor&& visitor) const {
-        // array should be already synchronized
-
-        Indexer<Ndim> indexer{array.shape()};
-        IndexableArray<const T, Ndim> iarray{array};
+    static constexpr int kMaxItemNumPerLine = 10;
 
-        for (auto it = indexer.It(0); it; ++it) {
-            visitor(iarray, it);
+    template <typename T>
+    void ArrayReprRecursive(const Array& array, Formatter<T>& formatter, size_t indent, std::ostream& os) const {
+        const uint8_t ndim = array.ndim();
+        if (ndim == 0) {
+            formatter.Print(os, static_cast<T>(AsScalar(array)));
```

Comment:
It seems you changed to use `static_cast` instead of `reinterpret_cast` which is the implementation of `native::StorageToDataType<const T>`. Is it intended fix? https://github.com/chainer/chainer/blob/ca48f1b8efbf980423631accb8264bd8bdfa8ccd/chainerx_cc/chainerx/native/data_type.h#L16-L18

Reply:
The original implementation uses `iarray`: `IndexableArray`, but I use `AsScalar` because `array` is guaranteed to be 0-dim array here.  `AsScalar` casts the data from `StorageType` to `DataType` internally, so we don't have to use `reinterpret_cast` here.
https://github.com/chainer/chainer/blob/b619124e1d88ce50a015c2a508184e843398f738/chainerx_cc/chainerx/routines/manipulation.cc#L48